app/schema/widgets/date_widget.py

```diff
@@ -80,11 +80,18 @@ def _get_year_field(self, year):
                 }
 
     @staticmethod
-    def _get_months(selected_month):
+    def _get_months(selected_month, is_mandatory):
         if selected_month:
             selected_month = int(selected_month)
 
         months = []
+
+        if not (selected_month and is_mandatory):
+            months.append(ObjectFromDict({
+                'value': '',
+                'text': 'Select month',
+            }))
+
```

Comment:
I did some research for the accessibility of the placeholder text for the drop down and found http://ux.stackexchange.com/questions/90642/accessibility-of-disabled-placeholders-in-dropdown. Following that advice and changing the `if not` block to the code below improves the usability of this for screen readers. The placeholder text is disabled for mandatory dates, but the option is selected until a month is chosen. 
```
        months.append(ObjectFromDict({
            'value': '',
            'text': 'Select month',
            'disabled': True if is_mandatory else False,
            'selected': False if selected_month else True
        }))
```